weavetest/auth.go

```diff
@@ -0,0 +1,76 @@
+package weavetest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/iov-one/weave"
+)
+
+// Auth is a mock implementing x.Authenticator interface.
+//
+// This structure authenticates any of referenced conditions.
+// You can use either Signer or Signers (or both) attributes to reference
+// conditions. This is for the convinience and each time all signers
+// (regardless which attribute) are considered.
+type Auth struct {
+	// Signer represents an authentication of a single signer. This is a
+	// convinience attribute when creating an authentication method for a
+	// single signer.
+	// When authenticating all signers declared on this structure are
+	// considered.
+	Signer weave.Condition
+
+	// Signers represents an authentication of multiple signers.
+	Signers []weave.Condition
+}
+
+func (a *Auth) GetConditions(weave.Context) []weave.Condition {
+	if a.Signer != nil {
+		return append(a.Signers, a.Signer)
+	}
+	return a.Signers
+}
+
+func (a *Auth) HasAddress(ctx weave.Context, addr weave.Address) bool {
+	for _, s := range a.Signers {
+		if addr.Equals(s.Address()) {
+			return true
+		}
+	}
+	return addr.Equals(a.Signer.Address())
+}
+
+// CtxAuth is a mock implementing x.Authenticator interface.
+//
+// This implementation is using context to store and retrieve permissions.
+type CtxAuth struct {
```

Comment:
nice doc strings, good useage (same as original?)

Reply:
New docs, implementation the same as before (`Key` type was changed to `string` though).